src/main/java/com/accelerex/tasks_manager/serviceImpl/UserServiceImpl.java

```diff
@@ -0,0 +1,278 @@
+        if (optionalUser.isPresent()) {
+
+            if (passwordEncoder.matches(dto.getPassword(), optionalUser.get().getPassword())) {
+                principalDTO.setMessage("Successful");
+                principalDTO.setSuccessful(true);
+                principalDTO.setToken(jwtUtil.generateToken(dto.getEmail()));
+            }
+        } else {
+            log.info("User not found");
+            principalDTO.setMessage("Not Successful");
+            principalDTO.setSuccessful(false);
+        }
+
+//        SecurityContextHolder.getContext().setAuthentication(authentication);
+
+
+        return principalDTO;
+
+    }
+
+    @Override
+    public UserDto updateUser(Long userId, UserDto userDto) {
+        Optional<User> optionalUser = userRepository.findById(userId);
+        if (optionalUser.isPresent()) {
+            User existingUser = optionalUser.get();
+            // Update the fields of the existing user with the values from userDto
+            existingUser.setFirstName(userDto.getFirstName());
+            existingUser.setLastName(userDto.getLastName());
+            existingUser.setEmail(userDto.getEmail());
+
+            // Save the updated user to the repository
+            User updatedUser = userRepository.save(existingUser);
+
+            // Convert the updated user entity to a UserDto and return it
+            UserDto updatedUserDto = new UserDto();
+            updatedUserDto.setId(updatedUser.getId());
+            updatedUserDto.setFirstName(updatedUser.getFirstName());
+            updatedUserDto.setLastName(updatedUser.getLastName());
+            updatedUserDto.setEmail(updatedUser.getEmail());
+
+            return updatedUserDto;
+        } else {
+            // Handle the case where user with given userId is not found
+            throw new UserNotFoundException("User with ID " + userId + " not found");
+        }
+    }
+
+    @Override
+    public void deleteUser(Long userId) {
+        Optional<User> optionalUser = userRepository.findById(userId);
+        if (optionalUser.isPresent()) {
+            User user = optionalUser.get();
+            userRepository.delete(user);
+        } else {
+            // Handle the case where user with given userId is not found
+            throw new UserNotFoundException("User with ID " + userId + " not found");
+        }
+    }
+
+    @Override
+    public User createStaff(StaffDto staffDto) {
+        log.info("Creating new staff member with email: {}", staffDto.getEmail());
+
+        Object user = SecurityContextHolder.getContext().getAuthentication().getPrincipal();
+        boolean isAdmin = false;
+        UserDetails userDetails = (UserDetails) user;
+        Optional<User> optionalUser1 = userRepository.findByEmail(userDetails.getUsername());
+        User user1 = optionalUser1.get();
+        Optional<UserRole> optionalUserRole = userRoleRepository.findById(user1.getRole().getId());
```

Comment:
you can actually extracted this into a different method say `isAdmiinUser()` to just check if it's an admin user so you reduce the responsibility if createStaff method.